android/src/main/java/carnegietechnologies/camera_content_saver/CameraContentSaverDelegate.java

```diff
@@ -0,0 +1,155 @@
+package carnegietechnologies.camera_content_saver;
+
+import android.Manifest;
+import android.app.Activity;
+import android.content.pm.PackageManager;
+import android.os.Handler;
+
+import androidx.core.app.ActivityCompat;
+
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel;
+import io.flutter.plugin.common.PluginRegistry;
+
+/**
+ * Class holding implementation of saving images and videos
+ */
+public class CameraContentSaverDelegate implements
+        PluginRegistry.RequestPermissionsResultListener {
+
+    private static final int REQUEST_EXTERNAL_IMAGE_STORAGE_PERMISSION = 2345;
+
+    private static final String ALREADY_ACTIVE = "already_active";
+    private static final String PLUGIN_ALREADY_ACTIVE = "camera content saver is already active";
+
+    private static final String KEY_PATH = "path";
+
+    private final Activity activity;
+
+    private MethodChannel.Result pendingResult;
+    private MethodCall methodCall;
+    private boolean isImage;
+
+    CameraContentSaverDelegate(Activity activity) {
+        this.activity = activity;
+    }
+
+
+    /**
+     * Saves image or video to device
+     *
+     * @param methodCall - method call
+     * @param result     - result to be set when saving operation finishes
+     * @param isImage    - tells if image or video should be saved
+     */
+    void saveFile(MethodCall methodCall, MethodChannel.Result result, boolean isImage) {
```

Comment:
why do you need to pass method call? Can we abstract method call away from this call and just pass path. Also, you have result as a member field. don't pass it since it's available in this scope.